src/components/Carousel.tsx

```diff
@@ -1,18 +1,119 @@
-import React from 'react';
+import { Component } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+};
+
+type State = {
+  position: number,
+};
+
+export default class Carousel extends Component<Props, State> {
+  state = {
+    position: 0,
+  };
+
+  clickNext = () => {
+    const { position } = this.state;
+    const {
+      images, step, itemWidth, frameSize, infinite,
+    } = this.props;
+    const maxPosition = images.length * itemWidth - frameSize * itemWidth;
+
+    if (infinite && position === -maxPosition) {
+      this.setState({ position: 0 });
+
+      return;
+    }
+
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    let rightMove = position - itemWidth * step;
+
+    rightMove = Math.max(rightMove, -itemWidth * (images.length - frameSize));
+
+    this.setState({ position: rightMove });
+  };
+
+  clickPrev = () => {
+    const { position } = this.state;
+    const {
+      step, itemWidth, infinite, images, frameSize,
+    } = this.props;
+    const leftMove = position + itemWidth * step;
+    const maxPosition = images.length * itemWidth - frameSize * itemWidth;
+
+    if (infinite && !position) {
+      this.setState({ position: -maxPosition });
+
+      return;
+    }
+
+    this.setState({ position: Math.min(leftMove, 0) });
+  };
+
+  render() {
+    const {
+      animationDuration, itemWidth, frameSize, infinite, images,
+    } = this.props;
+    const { position } = this.state;
+    const frameWidth = itemWidth * frameSize;
+    const maxPosition = images.length * itemWidth - frameSize * itemWidth;
+    const firstElem = infinite ? false : !position;
```

Comment:
```suggestion
    const isPrevButtonDisabled = infinite ? false : !position;
```
it's better to use longer but more meaningful variable names
boolean variable names should start with `is`/`has` etc.